modules/mill-codegen-plugin/test/src/smithy4s/codegen/mill/Smithy4sModuleSpec.scala

```diff
@@ -100,6 +100,24 @@ class Smithy4sModuleSpec extends munit.FunSuite {
       barEv.outPath / "smithy4sOutputDir.dest" / "scala" / "bar" / "Bar.scala",
       shouldExist = true
     )
+
+    val aScalaPath = foo.millSourcePath / "src" / "a.scala"
+
+    os.write(
+      aScalaPath,
+      """package foo
+      |object a""".stripMargin,
+      createFolders = true
+    )
+
+    // for some reason this works on mill
```

Comment:
...and it works without the fix. It appears that the file with `object a` does actually get used for compilation of `bar`, but I don't exactly understand why the `os.remove` isn't needed in this case.

It might be the case that Mill does a full clean of the generated directories anyway and I'm just duplicating work in that `os.remove` call. 

Might try with a manual test later, outside of the test infrastructure.

Reply:
ok, test managed to fail - all good!